src/unit-test/unit_test.py

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env python
+
+from src.text_files_splitter.text_files_splitter import TextFilesSplitter, ListMetrics
+import unittest
+
+
+class TestStringMethods(unittest.TestCase):
+
+    def test_csv(self):
```

Comment:
The tests should cover specific scenarios and specific functionality. This test only covers that you have a file to split and you get 4 files. It could be 4 empty files, 4 files with the wrong names, 4 files always because the max_size is being ignored, without headers, in the wrong output directory, etc. In this case, code coverage doesn't indicate code quality but it's misleading.

I recommend to create the following tests based on the functionality I've seen (maybe I'm missing more):

test_csv_is_splitted_in_4_files_of_10M
test_csv_is_splitted_in_3_files_of_15M
test_csv_adds_header_if_include_header_is_true
test_csv_does_not_add_header_if_include_header_is_false
test_csv_puts_files_in_output_directory
test_csv_uses_file_suffix_in_name_of_output_files

Reply:
You can generate dummy files for your tests.

To create dummy csv files on the fly you can use (replace *5 with the number you want to control how big the csv created is going to be):

```
import tempfile
import csv

handle, output = tempfile.mkstemp(suffix = ".csv")
writer = csv.writer(handle, delimiter=',', quoting=csv.QUOTE_MINIMAL)
writer.writerow(['header1', 'header2'])
writer.writerow([['a', 'b']] * 5)
return output
```

Now you can generate csv files to match all the scenarios that you need/want.

Reply:
Thank you so much Gorka, cool idea!